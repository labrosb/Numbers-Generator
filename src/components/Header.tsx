import React from 'react';
import { LinearGradient } from 'expo-linear-gradient';
import styled from 'styled-components/native';
import Layout from '../constants/Layout';
import colors from '../constants/Colors';
import Text from './Base/Text';

export const Container = styled(LinearGradient)`
  height: 110px;
  padding-top: ${Layout.statusBarHeight}px;
  justify-content: center;
  align-items: center;
  box-shadow: rgba(0, 0, 0, 0.14) 0px 2px 2px;
  elevation: 1.6;
`;

/** Presents the App Header */
const Header: React.FC<void> = () => (
  <Container
    colors={[colors.primary, colors.primaryTint]}
    start={[0, 1]}
    end={[1, 0]}
  >
    <Text size="18px" color={colors.secondary}>
      Numbers Generator
    </Text>
  </Container>
);

export default Header;
