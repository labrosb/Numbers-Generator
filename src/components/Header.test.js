import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Header from './Header';

configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Header />);
});

describe('Header component', () => {
  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
