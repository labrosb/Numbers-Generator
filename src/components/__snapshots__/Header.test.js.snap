// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Matches snapshot 1`] = `
<Styled(LinearGradient)
  colors={
    Array [
      "#576190",
      "#6d7099",
    ]
  }
  end={
    Array [
      1,
      0,
    ]
  }
  start={
    Array [
      0,
      1,
    ]
  }
>
  <Styled(Text)
    color="#FFF7E7"
    size="18px"
  >
    Numbers Generator
  </Styled(Text)>
</Styled(LinearGradient)>
`;
